File: backend/app/models/waste_assessment_history.py
from datetime import datetime
from pydantic import Field
from beanie import Indexed, Link
from app.models.base import BaseUUIDDocument
from app.models.cloud_resource import CloudResource
from app.models.base import get_utc_now

class WasteAssessmentHistory(BaseUUIDDocument):
    resource: Link[CloudResource]
    previous_score: float
    current_score: float
    risk_level: Indexed(str)
    assessment_date: Indexed(datetime) = Field(default_factory=get_utc_now)

    class Settings:
        name = "waste_assessment_history"
        indexes = [
            "assessment_date",
            "risk_level"
        ]

File: backend/src/models/WasteAssessmentHistory.ts
import mongoose, { Schema, Document } from 'mongoose';
import { BaseModelPlugin } from './BaseModelPlugin';
import crypto from 'crypto';
import { ICloudResource } from './CloudResource';

export interface IWasteAssessmentHistory extends Document<any> {
    _id: Buffer | string;
    resource: ICloudResource['_id'] | any;
    previous_score: number;
    current_score: number;
    risk_level: string;
    assessment_date: Date;
}

const wasteAssessmentHistorySchema = new Schema<IWasteAssessmentHistory>({
    _id: { type: 'UUID', default: () => crypto.randomUUID() },
    resource: { type: 'UUID', ref: 'CloudResource', required: true },
    previous_score: { type: Number, required: true },
    current_score: { type: Number, required: true },
    risk_level: { type: String, required: true, index: true },
    assessment_date: { type: Date, default: Date.now, index: true }
}, { collection: 'waste_assessment_history' });

wasteAssessmentHistorySchema.plugin(BaseModelPlugin);

export const WasteAssessmentHistory = mongoose.model<IWasteAssessmentHistory>('WasteAssessmentHistory', wasteAssessmentHistorySchema);
